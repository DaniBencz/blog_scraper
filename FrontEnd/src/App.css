#app {
}

#repo{
	margin: 7px;
}

#description{
	text-align: center;
}

#instructions {
	margin-top: 5px;
	color: white;
}

#form {
	text-align: center;
	background: #62d6a0;
	padding: 10px;
}

#page_value {
	width: 20px;
	font-size: 20px;
	text-align: center;
	margin-right: 10px;
}

#get_articles {
	font-size: 20px;
	margin-left: 10px;
	background: #3d87e2;
	color: white;
	border: none;
	padding: 3px 10px;
}

button:hover {
	cursor: pointer;
}

#loader{
	width: 200px;
	margin: auto;
}

ul {
	list-style-type: none;
}

#list {
	margin: auto;
	max-width: 800px;
}
