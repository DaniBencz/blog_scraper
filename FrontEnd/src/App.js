import React, { useState, useRef } from 'react';
//import logo from './logo.svg';
import './App.css';
import List from './components/List';
import axios from 'axios';

function App() {
	let pages = useRef(0);
	const [articles, setArticles] = useState([]);

	const updatePages = (event) => {
		pages = Number(event.target.value);
	}

	const getArticles = () => {
		axios({
			//url: `${window.location.origin}/articles`,
			url: 'http://localhost:4000/articles',
			method: 'post',
			data: {
				pages: pages
			}
		})
			.then(resp => {
				setArticles(resp.data.articles)
			})
			.catch(resp => {
				//
			})
	}

	return (
		<div>
			<input id="page" type="text" placeholder="type a number between 1 and 5" onChange={updatePages}></input>
			<button id="get_articles" onClick={getArticles}>Get Articles</button>
			<List articles={articles}></List>
		</div>
	);
}

export default App;
