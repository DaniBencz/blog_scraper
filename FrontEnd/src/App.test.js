import React from 'react'
import { render, fireEvent, queryByAttribute, wait, findByText } from '@testing-library/react'
import App from './App'
import axios from 'axios'

test('renders App', () => {
	const { getByText } = render(<App />)
	const welcome = getByText('Please type a number between 1 and 5, or leave empty!')
	expect(welcome).toBeInTheDocument()
})

jest.mock('axios')
const getById = queryByAttribute.bind(null, 'id')	// handy custom function

test('renders api response', async () => {

	let { container, getByText } = render(<App test={true} />)
	let resp = {
		data: {
			articles: ['hello']
		}
	}
	// axios.post.mockResolvedValue(resp)	// keeps returning undefined
	// axios.post.mockImplementation(() => Promise.resolve(resp))	// keeps returning undefined
	axios.mockReturnValue(Promise.resolve(resp))

	const input = getById(container, 'page_value')
	fireEvent.change(input, { target: { value: 1 } })

	const submit = getById(container, 'get_articles')
	fireEvent.click(submit)

	await wait(() => expect(axios).toHaveBeenCalledTimes(1))	// need to await async method getArticles
	await wait(() => expect(axios).toBeCalledWith({
		url: `${window.location.origin}/articles`,
		method: 'post',
		data: {
		  pages: 1
		}
	}))

	// expectations bellow do the same thing, with different error messege, if failing
	await wait(() => expect(getByText('hello')).toBeInTheDocument())
		// if fails, will complain for Timeout
	expect(await findByText(container, 'hello')).toBeInTheDocument()
		// if this fails, will give the right reason, unable to find element
})
