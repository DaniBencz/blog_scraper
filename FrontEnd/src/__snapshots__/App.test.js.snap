// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot test 1`] = `
<div
  id="app"
>
  <div
    id="repo"
  >
    <a
      href="https://github.com/DaniBencz/blog_scraper"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="gitHub repo"
        height="33"
        src="github_blue.png"
        width="33"
      />
    </a>
  </div>
  <div
    id="description"
  >
    <p>
      This is a practise project where 
      <a
        href="https://blog.risingstack.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        RisingStack blog
      </a>
       is being scraped for articles that contain no images (other than logos) or iFrames.
    </p>
  </div>
  <div
    id="form"
  >
    <h2
      id="instructions"
    >
      Please type a number between 1 and 5, or leave empty!
    </h2>
    <input
      id="page_value"
      onChange={[Function]}
      type="text"
    />
    <button
      id="get_articles"
      onClick={[Function]}
    >
      Get Articles
    </button>
  </div>
  <div
    id="list"
  >
    <ul />
  </div>
</div>
`;
